### PR TITLE
Adding regex check for handling failure in parsing dump files with specific syntax

### PR DESCRIPTION
**Pull request by shreyakhajanchi**
This regex check filters out the system generated code of the format :
/*!50717 SELECT COUNT(*) INTO @rocksdb_has_p_s_session_variables FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'performance_schema' AND TABLE_NAME = 'session_variables' */;

Such code format fails in parsing, we have added this regex as a workaround and will remove once pingcap provides a permanent fix for this : https://github.com/pingcap/parser/issues/1370

### CODE REVIEW

File: sources/mysql/mysqldump.go [hunk 1]
```diff
@@ -110,6 +114,10 @@ func readAndParseChunk(conv *internal.Conv, r *internal.Reader) ([]byte, []ast.S
 				n += copy(s[n:], l[i])
 			}
 			chunk := string(s)
+			matchStatus := regexExp.Match([]byte(chunk))
+			if matchStatus {
```
Comment (bharadwaj-aditya) [inline sources/mysql/mysqldump.go:118]: log that this is being skipped

File: sources/mysql/mysqldump.go [hunk 2]
```diff
@@ -94,6 +94,10 @@ func processMySQLDump(conv *internal.Conv, r *internal.Reader) error {
 // files containing tens or hundreds of GB of data.
 func readAndParseChunk(conv *internal.Conv, r *internal.Reader) ([]byte, []ast.StmtNode, error) {
 	var l [][]byte
+
+	// Regex for ignoring strings of the form /*!50717 SELECT COUNT(*) INTO @rocksdb_has_p_s_session_variables FROM INFORMATION_SCHEMA.TABLES */;
+	// These system generated SQL statements are currently not supported by parser and return error.
```
Comment (bharadwaj-aditya) [inline sources/mysql/mysqldump.go:99]: add reference to the bug in the comment

Comment (shreyakhajanchi) [reply sources/mysql/mysqldump.go:99]: added bug reference

File: sources/mysql/mysqldump.go (hunk 1 above)
Comment (shreyakhajanchi) [reply sources/mysql/mysqldump.go:119]: added log

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

File: sources/mysql/mysqldump.go (hunk 2 above)
Comment (bharadwaj-aditya) [reply sources/mysql/mysqldump.go:99]: this check will slow down the parsing, but let's make optimizing the parsing an independent activity

**Pull request merged**